app/helpers/samples_helper.rb

```diff
@@ -168,6 +168,25 @@ def parsed_samples_for_s3_path(s3_path, project_id, host_genome_id)
     sample_list
   end
 
+  def filter_samples(samples, params)
+    host = params[:host]
+    location = params[:location]
+    taxon = params[:taxon]
+    time = params[:time]
+    tissue = params[:tissue]
+    visibility = params[:visibility]
+
+    # # TODO(tiago): check performance
```

Comment:
in this PR?

Reply:
I actually checked already...
Still wanted to revisit once i have more time.